Makefile

```diff
@@ -65,14 +67,22 @@ lint-all: \
 
 VERSION=`echo "import accern_xyme;print(accern_xyme.__version__)" | python3 2>/dev/null`
 
-publish:
+git-check:
 	@git diff --exit-code 2>&1 >/dev/null && git diff --cached --exit-code 2>&1 >/dev/null || (echo "working copy is not clean" && exit 1)
 	@test -z `git ls-files --other --exclude-standard --directory` || (echo "there are untracked files" && exit 1)
 	@test `git rev-parse --abbrev-ref HEAD` = "master" || (echo "not on master" && exit 1)
+
+publish:
+	make git-check
 	rm -r dist build accern_xyme.egg-info || echo "no files to delete"
 	python3 -m pip install -U setuptools twine wheel
 	python3 setup.py sdist bdist_wheel
 	python3 -m twine upload dist/accern_xyme-$(VERSION)-py3-none-any.whl dist/accern_xyme-$(VERSION).tar.gz
 	git tag "v$(VERSION)"
 	git push origin "v$(VERSION)"
 	@echo "succesfully deployed $(VERSION)"
+
+publish-ts:
+	make git-check
+	yarn publish
```

Comment:
is there a way of retrieving the version that is in ts and check whether the tag exists?

Reply:
at least for non-rc?

Reply:
yarn will error if you publishing one version already exist

Reply:
git tag

Reply:
the version published must match the git tag